src/org/rascalmpl/parser/gtd/result/out/SkippedNodeFlattener.java

```diff
@@ -11,28 +11,67 @@
 *******************************************************************************/
 package org.rascalmpl.parser.gtd.result.out;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
 import org.rascalmpl.parser.gtd.location.PositionStore;
 import org.rascalmpl.parser.gtd.result.SkippedNode;
 
 /**
  * A converter for result nodes that contain skipped characters for error recovery
  */
 public class SkippedNodeFlattener<T, P>{
+	private static class MemoKey {
```

Comment:
Node caching will be solved more generically in the next PR. The memoization logic in this class will be removed completely.